test/type_list_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <unifex/type_list.hpp>
+
+using namespace unifex;
+
+void verify_unique_type_list_elements() {
+  static_assert(std::is_same_v<
+    unique_type_list_elements_t<type_list<>>,
+    type_list<>
+  >);
+  static_assert(std::is_same_v<
+    unique_type_list_elements_t<type_list<int>>,
+    type_list<int>
+  >);
+  static_assert(std::is_same_v<
+    unique_type_list_elements_t<type_list<int, int>>,
+    type_list<int>
+  >);
+  static_assert(std::is_same_v<
+    unique_type_list_elements_t<type_list<int, double, int>>,
+    type_list<double, int>
```

Comment:
Note that my hasty implementation of `unique_type_list_elements_t` does not preserve order, or rather, it removes duplicate items from the front rather than end of list.